hazelcast/src/main/java/com/hazelcast/sql/impl/expression/predicate/ComparisonPredicate.java

```diff
@@ -77,6 +79,12 @@ public Boolean eval(Row row, ExpressionEvalContext context) {
             return null;
         }
 
+        QueryDataTypeFamily leftTypeFamily = this.operand1.getType().getTypeFamily();
+        QueryDataTypeFamily rightTypeFamily = this.operand2.getType().getTypeFamily();
+        if (leftTypeFamily != rightTypeFamily) {
```

Comment:
It should never be the case that we have different type families here, because we use the coercion to convert two different types to a common type during the query optimization in `HazelcastComparisonPredicateUtils`.